packages/design-system/src/components/TextField/LabelMask/LabelMask.tsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+
+export enum BuiltInMask {
+  DATE = 'DATE',
+}
+
+export interface LabelMaskProps {
+  /**
+   * Must contain a `TextField` component
+   */
+  children: React.ReactNode;
+  labelMask?: BuiltInMask | ((rawInput: string) => string);
+}
+
+const LabelMask = ({ children, labelMask }: LabelMaskProps) => {
+  /**
+   * Get the child text field. Called as a method so that
+   * updates to the field cause the mask to re-render
+   * @returns {React.ReactElement} Child TextField
+   */
+  const field = (): React.ReactElement => React.Children.only(children as React.ReactElement);
+
+  /**
+   * @param {Object} evt
+   * @param {React.Element} field - Child TextField
+   */
+  const handleChange = (evt: React.ChangeEvent<HTMLInputElement>) => {
+    if (typeof field().props.onChange === 'function') {
+      return field().props.onChange(evt);
+    }
+  };
+
+  const modifiedTextField = React.cloneElement(field(), {
+    defaultValue: undefined,
+    onChange: (e) => handleChange(e),
+    type: 'text',
+    inputMode: 'numeric',
+  });
+
+  let hint = '';
+
+  function format(str): any {
+    if (labelMask === BuiltInMask.DATE) {
+      const charCount = str.length;
+
+      hint = 'MM/DD/YYYY';
+      const hintSub = hint.replace(/\//g, '').substring(charCount);
+
+      let val = str + hintSub;
+      // const dateRegex = /^(?<month>\d{1,2})(\s|-|\/)?(?<day>\d{0,2})(\s|-|\/)?(?<year>\d{0,4}).*/;
```

Comment:
I initially wanted to do something with named groups, since it makes the code more readable imo, but couldn't crack that one. I know there's a way to use names with `replace`. I wanted `str.replace(regex, "$<month>/$<day>/$<year>")` to work so badly :(